### PR TITLE
Open project wizard in modal from registration

### PR DESCRIPTION
**Pull request by denektenina**
# Description

Link to Jira issue: [https://sikt.atlassian.net/browse/NP-47743](https://sikt.atlassian.net/browse/NP-47743)

When making a new registration it should be possible to make a new project without having to leave the registration view (i.e. in a popup). This was possible earlier, so we are re-fetching old functionality in a new layout.

Once the project is created it is automatically added to the registration as well:

image

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/pages/project/ProjectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  toggleModal: () => void;
+  onProjectCreated: (value: CristinProject | ResearchProject) => void;
+}
+
+export const ProjectModal = ({ isOpen, toggleModal, onProjectCreated }: ProjectModalProps) => {
+  const { t } = useTranslation();
+
+  return (
+    <Modal
+      title={t('project.project_modal_title')}
```
Comment (elkebab) [inline src/pages/project/ProjectModal.tsx:17]: Får to a11y errors her pga manglende heading i dialogen. Ser ut til å la seg løse ganske enkelt ved å endre prop

```suggestion
      headingText={t('project.project_modal_title')}
```

bilde

File: src/pages/project/ProjectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      open={isOpen}
+      onClose={toggleModal}
+      fullWidth
+      maxWidth="lg"
+      dataTestId="contributor-modal">
```
Comment (elkebab) [inline src/pages/project/ProjectModal.tsx:22]: dataTestId skulle sikkert vært fra `dataTestIds` og hete noe med prosjekt?

File: src/pages/project/project_wizard/ProjectForm.tsx [hunk 3]
```diff
@@ -28,9 +28,11 @@ import { ProjectFormStepper } from './ProjectFormStepper';
 interface ProjectFormProps {
   project: SaveCristinProject | CristinProject;
   suggestedProjectManager?: string;
+  toggleModal?: () => void;
+  onProjectCreated?: (value: CristinProject | ResearchProject) => void;
```
Comment (elkebab) [inline src/pages/project/project_wizard/ProjectForm.tsx:32]: ser ut som dette bare kan være `CristinProject`?
```suggestion
  onProjectCreated?: (value: CristinProject) => void;
```

**Review by elkebab — CHANGES_REQUESTED** [review]
Noen småting med a11y, ellers syns jeg egentlig det føltes ganske bra i bruk!

File: src/pages/project/project_wizard/ProjectForm.tsx [hunk 4]
```diff
@@ -132,6 +145,7 @@ export const ProjectForm = ({ project, suggestedProjectManager }: ProjectFormPro
                   onClickNext={() => onClickArrow(tabNumber + 1)}
                   onClickPrevious={() => onClickArrow(tabNumber - 1)}
                   onClickLast={() => onClickArrow(ProjectTabs.Connections)}
+                  openedInModal={toggleModal !== undefined}
```
Comment (elkebab) [inline src/pages/project/project_wizard/ProjectForm.tsx:148]: for å matche sjekkene på linje 68 og 88
```suggestion
                  openedInModal={toggleModal}
```

File: src/pages/project/ProjectModal.tsx (hunk 1 above)
Comment (denektenina) [reply src/pages/project/ProjectModal.tsx:17]: Tar den heller bort

Comment (elkebab) [reply src/pages/project/ProjectModal.tsx]: Tror du må ha den? Jeg får fortsatt to feil for a11y etter testing med axe hvertfall (1 serious og 1 minor)

**Review by elkebab — APPROVED** [review]
Blir jo litt ekkel håndtering av om det er modal eller ikke rundt omkring, men thats life

**Pull request merged**